File: Semester 4/Full Stack Development/Lab Assessment 3/05.js
// Write a function manipulateArray(arr, operations) that takes an array and a list of operations. Each operation will be a string "filter", "map", or "reduce".
// ●	"filter" should filter out numbers less than 10.
// ●	"map" should multiply each element by 2.
// ●	"reduce" should sum all elements.
// Demonstrate your function with [3, 15, 20, 8, 5, 12] and the operation sequence ["filter", "map", "reduce"].


function manipulateArray(arr, operations) {
    let result = arr;

    operations.forEach(operation => {
        if (operation === "filter") {
            result = result.filter(num => num < 10);
        } else if (operation === "map") {
            result = result.map(num => num * 2);
        } else if (operation === "reduce") {
            result = result.reduce((acc, num) => acc + num, 0);
        }
    });

    return result;
}

const arr = [3, 15, 20, 8, 5, 12];
const operations = ["filter", "map", "reduce"];

const result = manipulateArray(arr, operations);
console.log(result);

File: Semester 5/Self/02-DSA/08 - 2D Arrays/05.cpp
#include <bits/stdc++.h>
using namespace std;

int diagonalSum(int arr[][3], int n){
    int sum = 0;

    // for(int i=0; i<n; i++){
    //     for(int j=0; j<n; j++){
    //         if(i==j){
    //             sum+=arr[i][j];
    //         }
    //         else if(j == n-i-1){
    //             sum+=arr[i][j];
    //         }

    //     }
    // }

    for(int i=0; i<n; i++){
        sum += arr[i][i];

        if(i!=n-i-1){
            sum+=arr[i][n-i-1];
        }
    }

    return sum;
}

int main() {
    int m1[4][4] = {{1,2,3,4}, {5,6,7,8}, {9,10,11,12}, {13,14,15,16}};
    int m2[3][3] = {{1,2,3}, {4,5,6}, {7,8,9}};

    cout << diagonalSum(m2, 3);
    return 0;
}